Ubuntu base

cdbishop opened the pull request:
@mattyclarkson ok If I merge this?
I've tested this locally running a tool-chain and it configures and builds fine.

mattyclarkson committed:
Use ubuntu as a base

This image is optimised for usage with Docker, we can convert it into
Linux Mint by updating the repositories and packages

mattyclarkson committed:
Set up the Linux Mint repositories

This adds the Linux Mint repository and makes sure that the key is
added to the keyring so that packages are correctly verified

mattyclarkson committed:
Install base-files

This installs the necessary /etc/lsb-release to make the image
think that it is Linux Mint

mattyclarkson commented:
@cdbishop, do we need to do your update/upgrade after the install of the `base-files`? I think I actually tried that and it didn't seem to update anything. Happy for merge though if it works in the TC! yay, saveing about 1.4GB!

mattyclarkson commented on Dockerfile:
```diff
@@ -1,4 +1,4 @@
-FROM vcatechnology/base-linux-mint
+FROM vcatechnology/ubuntu
```
Ooooo, need to put `:16.04` here to make sure we don't break!

cdbishop replied:
Oh yea. Done.

cdbishop committed:
Force derived ubuntu base to be 16.04

cdbishop committed:
Update linux mint packages via dist-upgrade

We need the APT::Immediate-Configure=false to avoid circular
configuration problems when upgrading to the Linux Mint packages.

cdbishop committed:
Add the backport repository

mattyclarkson commented:
OK, merged. Have updated the `18` branch to point to `master`. Need to implement the `17.3` branch. Which should be pretty simple.
